### ISSUE TITLE
Support DIN 5007-2 sorting

### ISSUE THREAD

**Issue by timohund** [TYPO3-Solr/ext-solr#535]
German umlauts ä/ö/ü are sorted after their equivalents ae/oe/ue.
What we need is an out-of-the box solution to achieve that.

For more information, see http://wiki.apache.org/solr/UnicodeCollation section "Sorting text with custom rules"

https://forge.typo3.org/issues/52489

**Comment by timohund** [comment]
[din5007-2.zip](https://github.com/TYPO3-Solr/ext-solr/files/366266/din5007-2.zip)

**Comment by timohund** [comment]
**1**

 Updated by Ingo Renner almost 3 years ago

Added customRules.dat

**2**

Updated by Ingo Renner almost 3 years ago
Comment Edit

Add a "collated" sorting field to types and fields, have types specific for each language, and a general for languages where a collation is not available.

```
<fieldType name="collated" class="solr.TextField">
    <analyzer>
        <tokenizer class="solr.KeywordTokenizerFactory"/>
        <filter class="solr.CollationKeyFilterFactory" 
            language="de" 
            strength="primary" />
    </analyzer>
</fieldType>
```

**3**
Updated by Ingo Renner about 2 years ago


Comment Edit

Moving this out to 3.1.

Spent some time now trying to add this to the schema, but Solr fails to initialize the schema. It seems the customRules.dat file attached here has been corrupted (at least Solr complains about it). Need to regenerate the rules. While at it might also check whether DIN5007-2 treats ß == ss and then add that, too.

solr.CollationKeyFilterFactory meanwhile has been deprecated. solr.ICUCollationField should be used now, see https://cwiki.apache.org/confluence/display/solr/Language+Analysis#LanguageAnalysis-UnicodeCollation

The rules file simply needs to be added to typo3cores/conf/

**Comment by dkd-kaehm** [comment]
can be closed, isn't it done in #1571?
Please reopen if wrongly closed!
